compiler/src/dotty/tools/dotc/ast/Positioned.scala

```diff
@@ -23,11 +23,14 @@ abstract class Positioned(implicit @constructorOnly src: SourceFile) extends Pro
    */
   def uniqueId: Int = myUniqueId
 
-  private def uniqueId_=(id: Int): Unit = {
-    // FOR DEBUGGING
-    // assert(id != 2523, this)
-    // if (id == 1234) new Throwable().printStackTrace()
-    
+  def uniqueId_=(id: Int): Unit = {
+    if (Positioned.debugId == id) {
+      def printTrace() = {
+        val stack = Thread.currentThread().getStackTrace().map(">   " + _)
+        System.err.println(stack.mkString(s"> Debug tree (id=${Positioned.debugId}) creation \n> $this\n", "\n", "\n"))
+      }
+      printTrace()
```

Comment:
Why is the logic abstracted into a method instead of inlining it?

Reply:
`uniqueId_=` is probably a hot path as each tree creation will have a call to it. I put the code contained in this debug branch to make the bytecode of `uniqueId_=` smaller and probably simple to optimize from the JVM perspective.